sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/FailureSafeParser.scala

```diff
@@ -40,14 +40,20 @@ class FailureSafeParser[IN](
   // set the bad record to this field, and set other fields according to the partial result or null.
   private val toResultRow: (Option[InternalRow], () => UTF8String) => InternalRow = {
     (row, badRecord) => {
-      var i = 0
-      while (i < actualSchema.length) {
-        val from = actualSchema(i)
-        resultRow(schema.fieldIndex(from.name)) = row.map(_.get(i, from.dataType)).orNull
-        i += 1
+      // save the value. Some implementations of badRecord do not like to be called twice
+      val badRec = badRecord()
+      if (badRec != null || corruptFieldIndex.isDefined) {
```

Comment:
I think partial result should be returned even the corrupt column is not defined.

Reply:
@MaxGekk 
>I think partial result should be returned even the corrupt column is not defined.

It works that way (I hope). For example, id2 is a DateType field that contains bad data:
<pre>
scala> sql("select id1, id2 from csvtbl").show
sql("select id1, id2 from csvtbl").show
+-----+----+
|  id1| id2|
+-----+----+
|276.0|null|
|176.0|null|
|  2.0|null|
|355.0|null|
|263.0|null|
|172.0|null|
|196.0|null|
|352.0|null|
....etc...
</pre>
No corrupt record field was specified in this case.

I tried not to change the semantics of the partial result changes.